src/Json/Serializer.ts

```diff
@@ -1,50 +1,55 @@
 import {TypeUtil} from "../Utility/TypeUtil";
 import {DateUtil} from "../Utility/DateUtil";
 import {ArrayUtil} from "../Utility/ArrayUtil";
+import {INestedObjectTypes, DocumentConstructor} from "../Documents/Conventions/DocumentConventions";
 
 export class Serializer {
-  public static fromJSON<T extends Object>(className: { new(): T; }, source: Object | string, metadata: Object = {}, target?: T)
+  public static fromJSON<T extends Object>(target: T, source: Object | string, metadata: Object = {}, nestedObjectTypes: INestedObjectTypes = {}): T
```

Comment:
If we're gonna do it this way - construct objects from results using constructors etc. , which I am not convinced we should - this class should have a solid set of tests.